sys-utils/lscpu.c

```diff
@@ -999,11 +1000,11 @@ static void print_summary(struct lscpu_cxt *cxt)
 	}
 	if (ct && ct->addrsz)
 		add_summary_s(tb, sec, _("Address sizes:"), ct->addrsz);
-#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
-	add_summary_s(tb, sec, _("Byte Order:"), "Little Endian");
-#else
-	add_summary_s(tb, sec, _("Byte Order:"), "Big Endian");
-#endif
+
+	if (procfs_get_byteorder(cxt->procfs) == PROCFS_BYTEORDER_LITTLE)
+		add_summary_s(tb, sec, _("Byte Order:"), "Little Endian");
+	else
+		add_summary_s(tb, sec, _("Byte Order:"), "Big Endian");
```

Comment:
That's nice because it will also work for /proc dumps (used, for example, in lscpu regression tests).

Reply:
That was intentional 😃